.github/workflows/pull_request.yml

```diff
@@ -131,22 +142,24 @@ jobs:
           cache: "npm"
       - run: |
           echo "PYTHON_VERSION=$(cat .python-version)" >> $GITHUB_ENV
-      - uses: actions/setup-python@v4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.8.2
+          virtualenvs-create: true
+      - uses: actions/setup-python@v5
+        id: cache-poetry
```

Comment:
id is now redundant, hence can be removed.

Reply:
Removed